### PR TITLE
Store waiting transactions in DB

### PR DESCRIPTION
**Pull request by palango**
@karlb Can you have a first look on if this looks reasonable

Part of #157

### CODE REVIEW

File: src/monitoring_service/schema.sql [hunk 1]
```diff
@@ -52,3 +52,7 @@ CREATE TABLE monitor_request (
     --FOREIGN KEY (channel_identifier, token_network_address)
     --    REFERENCES channels(channel_identifier, token_network_address) ON DELETE CASCADE
 );
+
+CREATE TABLE waiting_transactions (
+    transaction_hash        CHAR(66)    NOT NULL
```
[the remark this answers is not in the document]
Comment (palango) [reply src/monitoring_service/schema.sql]: That would only be beneficial when we select certain hashes but not when we get all of them, right?

File: src/monitoring_service/database.py [hunk 2]
```diff
@@ -143,6 +143,25 @@ def get_channel(self, token_network_address: str, channel_id: int) -> Optional[C
...
+    def get_waiting_transactions(self) -> List[str]:
+        return [
+            row[0] for row in self.conn.execute(
+                "SELECT transaction_hash FROM waiting_transactions",
+            )
+        ]
+
+    def add_waiting_transaction(self, waiting_tx_hash: str) -> None:
+        self.conn.execute(
+            "INSERT OR REPLACE INTO waiting_transactions VALUES (?)",
```
[the remark this answers is not in the document]
Comment (palango) [reply src/monitoring_service/database.py]: The hashes should be unique, so I guess it's not necessary. Does it hurt though and/or should we add a unique constraint to the table definition?

File: src/monitoring_service/database.py [hunk 3]
```diff
@@ -143,6 +143,27 @@ def get_channel(self, token_network_address: str, channel_id: int) -> Optional[C
     def channel_count(self) -> int:
         return self.conn.execute("SELECT count(*) FROM channel").fetchone()[0]
 
+    def get_waiting_transactions(self) -> List[str]:
```
Comment (BOR4) [inline src/monitoring_service/database.py]: if this list tends to get big maybe you should consider implementing iterator (in the future).

Comment (palango) [reply src/monitoring_service/database.py]: Good point, but this list should never get that big.

File: src/monitoring_service/schema.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (BOR4) [reply src/monitoring_service/schema.sql]: yes primary key afaik creates index and that helps in search :)

[the remark this answers is not in the document]
Comment (karlb) [reply src/monitoring_service/schema.sql]: Regarding performance, yes. But it also prevents inserting the same hash twice.

**Pull request merged**